Porting across bloom documentation (#2375)
ros2/ros2_documentation: source/How-To-Guides/Releasing/Releasing-a-Package.rst

.. redirect-from::

    Releasing-a-ROS-2-package-with-bloom
    Guides/Releasing-a-ROS-2-package-with-bloom
    Tutorials/Releasing-a-ROS-2-package-with-bloom
    How-To-Guides/Releasing-a-ROS-2-package-with-bloom

Releasing a Package
===================

.. toctree::
   :hidden:

   First-Time-Release
   Subsequent-Releases
   Release-Team-Repository
   Release-Track
   GitHub-Manual-Authorization

> **Releasing a package in ROS 2 means to make your package available on the public ROS 2 buildfarm.**

Reviewer: ```suggestion
**Releasing a package makes your package available on the public ROS 2 buildfarm.**
```

> After you've created a package, this is the next step towards getting your package into the publicly-available Debian packages so that others can install the package via ``apt``.

Reviewer: ```suggestion
After you've created a package, this is the next step towards getting your package into the publicly available Debian packages so that users can install the package via ``apt``.
```

Reviewer: Actually, while this sentence is true, it isn't the whole story.

Releasing your package onto the buildfarm will do the following things:

1.  Make your package available to be installed by package managers *for all of the currently supported Linux platforms* for that ROS distributions.  That currently means Ubuntu 22.04 and RHEL 8 for Rolling, but may expand or contract in the future.
2. (optionally) Allows you to have automatic CI run for pull requests to your repository.
3. Allows your package to have API documentation automatically generated.
4. Makes your package part of the index at https://index.ros.org .

I think it is worthwhile expanding this bit to talk about these advantages a bit more.

Author: Thanks for listing these out, I've added it in

.. redirect-from::

    Releasing-a-ROS-2-package-with-bloom
    Guides/Releasing-a-ROS-2-package-with-bloom
    Tutorials/Releasing-a-ROS-2-package-with-bloom
    How-To-Guides/Releasing-a-ROS-2-package-with-bloom

Releasing a Package
===================

.. toctree::
   :hidden:

   First-Time-Release
   Subsequent-Releases
   Release-Team-Repository
   Release-Track

**Releasing a package makes your package available on the public ROS 2 buildfarm.**

> After you have created a package, this is the next step towards getting your package into the publicly available Debian packages!

Reviewer: I think we can just remove this sentence now, as it is redundant with the next one.

Author: This should be good now

.. redirect-from::

    Releasing-a-ROS-2-package-with-bloom
    Guides/Releasing-a-ROS-2-package-with-bloom
    Tutorials/Releasing-a-ROS-2-package-with-bloom
    How-To-Guides/Releasing-a-ROS-2-package-with-bloom

Releasing a Package
===================

.. toctree::
   :hidden:

   Obtain-Access-to-Release-Repository
   Releasing-an-Ament-Package
   Releasing-a-Third-Party-Package
   Github-Manual-Authorization

Releasing a package in ROS2 means to make your package available on the public ROS 2 buildfarm.
After you've created a package, this is the next step towards getting your package in to the
publicly-available Debian packages so that others can install the package via ``apt``.

First, **follow instructions to** :doc:`Obtain Access to Release Repository <Obtain-Access-to-Release-Repository>`.

Second, **select one of the following**, according to the type of package you are releasing:

* :doc:`Releasing an Ament Package <Releasing-an-Ament-Package>`

   A package which contains a ``package.xml`` and follows the packaging guidelines
   of ament. Most ROS packages are ament packages.
   

> * :doc:`Releasing an Ament Package <Releasing-an-Ament-Package>`

Reviewer 2: I believe this one should be
```suggestion
* :doc:`Releasing an Ament Package <Releasing-a-Third-Party-Package>`
```

Author: Thank for picking this up, fixed this in https://github.com/ros2/ros2_documentation/pull/2375/commits/2748a3c791d36cf632e10a338608b9c2954983c7
